tests/foreman/api/test_templatesync.py

```diff
@@ -1056,3 +1021,187 @@ def test_positive_export_log_to_production(
             ).status
             == 0
         )
+
+    @pytest.mark.tier2
+    @pytest.mark.skip_if_not_set('git')
+    @pytest.mark.parametrize(
+        'url',
+        [
+            f'http://{git.username}:{git.password}@{git.hostname}:{git.http_port}',
+            f'ssh://git@{git.hostname}:{git.ssh_port}',
+        ],
+        ids=['http', 'ssh'],
+    )
+    @pytest.mark.parametrize(
+        'repo, branch',
+        [
+            (lazy_fixture(('git_repository', 'git_branch'))),
```

Comment:
I don't think this dependency is strictly required for what you're trying to do here.

Whether the created repository has `init=True` or `init=False` within the fixture could be indirectly parametrized.
The branch can also be indirectly parametrized.

Reply:
But this would mean that every time test is created it will also create new repository or not?

Reply:
The scope of the fixtures still applies when indirectly parametrizing, in this case you have session scoped fixtures (I'd recommend module scope since they're defined in the local test module).